ui/core/Button.tsx

```diff
@@ -8,13 +8,21 @@ interface Props {
   children: React.ReactNode
   left?: React.ReactNode
   right?: React.ReactNode
+  dataAttributes?: {
+    [id: string]: string
+  }
 }
 type ButtonProps = JSX.IntrinsicElements['button'] & Props & As
 
 export const Button: React.FC<ButtonProps> = React.forwardRef(
-  ({ children, left, right, ...rest }, ref) => {
+  ({ children, left, right, dataAttributes, ...rest }, ref) => {
+    const data = {}
+    Object.keys(dataAttributes || {}).forEach(function (key) {
+      data[`data-${key}`] = dataAttributes[key]
+    })
```

Comment:
```suggestion
      const data = Object.entries(dataAttributes || {})
        .reduce((data, [key, value]) => ({ [`data-${key}`]: value, ...data }), {})
    )
```

Reply:
This would be the functional way to do this in javascript, I'll leave it up to you if this is more or less readable :smile: